Objects/unionobject.c

```diff
@@ -494,6 +494,29 @@ static PyNumberMethods union_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static const char* const cls_attrs[] = {
+        "__name__",
+        "__qualname__",
+        "__module__",
+        NULL,
+};
+
+static PyObject *
+union_getattro(PyObject *self, PyObject *name)
+{
+    unionobject *alias = (unionobject *)self;
+
+    for (const char * const *p = cls_attrs; ; p++) {
```

Comment:
Please add a ``PyUnicode_Check(name)`` around this for loop.

`PyObject_GetAttr` and `PyObject_GenericGetAttr` can both handle non-unicode and raise error. But `_PyUnicode_EqualToASCIIString` does no checks at all in release builds.